state/core_access.go

```diff
@@ -265,15 +254,15 @@ func (ca *CoreAccessor) SubmitPayForBlob(
 			continue
 		}
 
-		// metrics should only be counted on a successful PFD tx
-		if err == nil && response.Code == 0 {
-			ca.markSuccessfulPFB()
+		if err != nil {
+			return nil, err
 		}
 
-		if response != nil && response.Code != 0 {
-			err = errors.Join(err, sdkErrors.ABCIError(response.Codespace, response.Code, response.Logs.String()))
```

Comment:
Could you please explain why this error construction is removed? Looks like a breaking change for errors returned to client in this particular case

Reply:
> Errors should be properly constructed and returned by the tx client itself

The node is a tx client. Doesn't that mean that this change should construct the error now?

Reply:
I'm implying the tx client which lives in app that node uses. See one of the examples of how we handle the errors now here. https://github.com/celestiaorg/celestia-app/blob/48173df3dc78f9348eedb3796f29ef9e9e5dc584/pkg/user/tx_client.go#L381